sql_5.sql

```diff
@@ -0,0 +1,205 @@
+     FOREIGN KEY(theatreid) REFERENCES theatres(theatreid),
+  )
+  CREATE TABLE shows
+  (
+	 showid  INT PRIMARY KEY IDENTITY(1, 1),
+	 screenid INT,
+	 movieid  INT,
+	 time	DATETIME,	 
+     FOREIGN KEY(screenid) REFERENCES screens(screenid),
+     FOREIGN KEY(movieid) REFERENCES movies(movieid)
+  )
+
+  
+CREATE TABLE language
+  (	 
+	 languageid  INT PRIMARY KEY IDENTITY(1, 1),
+	 language	 VARCHAR(25)	
+  )
+
+CREATE TABLE movielanguage
+  (
+	 movielanguageid  INT PRIMARY KEY IDENTITY(1, 1),
+	 languageid		  INT,
+	 movieid		  INT,
+     FOREIGN KEY(languageid) REFERENCES language(languageid),
+     FOREIGN KEY(movieid) REFERENCES movies(movieid)
+  )
+     
+
+
+CREATE TABLE orderdetails
+  (
+	 orderdetailsid INT PRIMARY KEY IDENTITY(1, 1),
+     customerid     INT,
+     theatreid      INT,
+     movieid        INT,
+     orderdatetime  DATETIME,
+     showdatetime   DATETIME,
+     amount         DECIMAL(10, 2),
+     paymentmethod  VARCHAR(25),
+     FOREIGN KEY(theatreid) REFERENCES theatres(theatreid),
+     FOREIGN KEY(customerid) REFERENCES customers(customerid),
+     FOREIGN KEY(movieid) REFERENCES movies(movieid)
+  ) 
+
+  CREATE TABLE seats
+  (
+	 seatid         INT PRIMARY KEY IDENTITY(1, 1),
+	 screenid		INT,
+	 seatnumber		INT,
+	 isavailable   BIT DEFAULT 1,
+	 price			DECIMAL(10, 2),
+     FOREIGN KEY(screenid) REFERENCES screens(screenid),
+  )
+
+  CREATE TABLE bookedseats
+  (
+	 bookedseatsid  INT PRIMARY KEY IDENTITY(1, 1),
+	 seatid		    INT,
+	 orderdetailsid INT,
+     FOREIGN KEY(seatid) REFERENCES seats(seatid),
+     FOREIGN KEY(orderdetailsid) REFERENCES orderdetails(orderdetailsid)
+  )
+
+  
+------------------------------------------------------------------------------------------------------------------------
+amazon
+
+
+CREATE TABLE customers
+  (
+     customerid      INT PRIMARY KEY IDENTITY(1, 1),
+     customername    VARCHAR(25),
+     customeremail   VARCHAR(25) UNIQUE,
+     customerphoneno VARCHAR(13),
+     address         VARCHAR(100),
+  )
+
+CREATE TABLE sellers
+  (
+     sellerid      INT PRIMARY KEY IDENTITY(1, 1),
+     sellername    VARCHAR(25),
+     selleremail   VARCHAR(25) UNIQUE,
+     sellerphoneno VARCHAR(13),
+     address         VARCHAR(100),
+  )
+
+CREATE TABLE products
+  (
+     productid   INT PRIMARY KEY IDENTITY(1, 1),
+     productname VARCHAR(25),
+     category    VARCHAR(25),
+     price       DECIMAL(10, 2)
+  )
+
+CREATE TABLE orders
+  (
+     orderid           INT PRIMARY KEY IDENTITY(1, 1),
+     customerid        INT,
+     sellerid          INT,
+     orderdate         DATE,
+     FOREIGN KEY(customerid) REFERENCES customers(customerid),
+     FOREIGN KEY(sellerid) REFERENCES sellers(sellerid)
+  )
+
+CREATE TABLE orderdetails
+  (
+     orderdetailsid INT PRIMARY KEY IDENTITY(1, 1),
+     orderid        INT,
+     productid      INT,
+     quantity       INT,
+	 amount         DECIMAL(10, 2),
+	 paymentmethod  VARCHAR(20),
+	 orderstatus	VARCHAR(20),
```

Comment:
if someone buys more than one products on a single order then order status of each product could be tracked

Reply:
for that purpose we can add an orderstatus in the orders table to show whether the orders are completely delivered or not and for each products order status it can be given in orderdetails table as ProductStatus. Suppose if i buy a phone and a t shirt from amazon the each product will have their own order status one may get shipped earlier than other or one may get delivered earlier than other.

Reply:
also we can change the seller id from orders table to orderdetails table if each product is sold by a different seller